docs/testing.md

```diff
@@ -36,3 +36,46 @@ Run linters test before pushing your commit
 ```bash
 make lint
 ```
+
+## Using the Hack scripts
+The repository contains a ``hack`` directory which has some very useful scripts
+for setting up testing environments for various purposes. It also has Dockerfiles to
+allow you to setup your linter/tester containers easily.
+
+All the scripts accept two environment variables
+
+1. ``IS_CONTAINER`` default: false; specify if you want tests run inside a container
+2. ``CONTAINER_RUNTIME`` default: podman; specify the runtime you want to use
+
+As you can see, by default, the tests will not run in a container and if chosen to
+be containerized, will do so with podman, you can tweak these variables to your liking.
+
+### Example of running golint in a docker container
+Start by setting the environment variables appropriately
+
+```bash
+export IS_CONTAINER=true
```

Comment:
you should not set that. If you set it, it will not run as a container, it will run locally. It is just a flag so that docker or podman would run the same script, but identify that it is running as a container already.

Reply:
I think we should prefer people use `make` if they have the tools installed and don't need the containerized versions. Otherwise they should just run the script, and optionally set the `CONTAINER_RUNTIME`, but never set `IS_CONTAINER`.

Reply:
Oh, got it. Thanks! Sending a patch now.